uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/columns/impl/IntegerColumnRenderer.java

```diff
@@ -27,20 +27,29 @@
     @Override
     public Group renderCell(final GridCell<Integer> cell,
                             final GridBodyCellRenderContext context) {
-        if (cell == null || cell.getValue() == null || cell.getValue().getValue() == null) {
+        if (cell == null || cell.getValue() == null || (cell.getValue().getValue() == null && cell.getValue().getPlaceHolder() == null)) {
             return null;
         }
 
         final GridRenderer renderer = context.getRenderer();
         final GridRendererTheme theme = renderer.getTheme();
 
-        final Group g = new Group();
-        final Text t = theme.getBodyText();
-        t.setText(Integer.toString(cell.getValue().getValue()));
-        t.setListening(false);
-        t.setX(context.getCellWidth() / 2);
-        t.setY(context.getCellHeight() / 2);
-        g.add(t);
-        return g;
+        Text text;
+        String value;
+
+        // Show placeholder only if the following conditions are met
+        if (cell.getValue() != null && cell.getValue().getValue() == null && cell.getValue().getPlaceHolder() != null) {
```

Comment:
I would add a method here ( and in `StringColumnRenderer.java` ) that removes the comment above. Something like:
`if(isCellEmpty(cell))`
Just to remove the comment above and mention why the placeholder used.

And also test this if so that the next guy does not come and break your code.